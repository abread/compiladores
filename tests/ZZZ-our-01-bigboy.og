auto f(int a) {
  auto b = 1, 2, 3, a;
  auto d = (2, 4), b, b;

  if ((((a)))) then {
    return 5, ((((61.0)))), (("string1")), d;
  } else {
    return (((((6, 62, "string2", ((3.1, 5.1), (2.1, 3.1, 4.1, -4.2), (-4.2, 4.1, 3.1, 2.1)))))));
  }
}
public int og() {
  auto f0, f1 = f(0), f(1);

  auto a, b, c, d = f0;
  auto d1 = d@1;
  auto d2 = d@2;
  auto d3 = d@3;
  auto inner = d@2@4;

  (1,2.2,3)@2 = 2;

  writeln a;
  writeln b;
  writeln c;

  writeln d1@1;
  writeln d1@2;
  writeln d2@1;
  writeln d2@2;
  writeln d2@3;
  writeln d2@4;
  writeln d3@1;
  writeln d3@2;
  writeln d3@3;
  writeln d3@4;

  writeln inner;
  writeln f1@4@3@4 * (nullptr, (3.4, ("ola", 6.1), "asd"), 4)@2@2@2 / 2 + 3;
  return 0;
}
