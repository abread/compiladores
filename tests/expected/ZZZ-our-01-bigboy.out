6
6.2E1
string2
3.1
5.1
2.1
3.1
4.1
-4.2
-4.2
4.1
3.1
2.1
-4.2
6.05